[Security] Document the new `expose_security_errors` option (#21008)
symfony/symfony-docs: reference/configuration/security.rst

* `access_denied_url`_
* `erase_credentials`_

> * `hide_user_not_found`_

Reviewer: shouldn't we keep it here and maybe sufis it with ` (deprecated)` ?

Author: Yes! Let's do that.

   Since Symfony 7.3, ``eraseCredentials()`` methods are deprecated and are
   not called if they have the ``#[\Deprecated]`` attribute.

expose_security_errors
----------------------

**type**: ``string`` **default**: ``'none'``

.. deprecated:: 7.3

    The ``expose_security_errors`` option was introduced in Symfony 7.3

> User enumeration is a common security issue where attackers infer valid usernames

Reviewer: maybe adding a link to some page about user enumeration?

Author: I can't find a link in Wikipedia and all the other links are not great. I'd say it's enough with the short given description.

based on error messages. For example, a message like "This user does not exist"

> shown by your login form reveals whether a username is valid.

Reviewer: ```suggestion
shown by your login form reveals whether a username exists, or not.
```

Author: I changed this, but removed the `or not` because all AIs told me it's better to not include that.
